### PR TITLE
Implement syntax and parsing for value generics

### PR DESCRIPTION
**Pull request by Azoy**
This is to accommodate the language feature being developed here: https://github.com/swiftlang/swift/pull/75518

### CODE REVIEW

File: CodeGeneration/Sources/SyntaxSupport/GenericNodes.swift [hunk 1]
```diff
@@ -93,10 +93,12 @@ public let GENERIC_NODES: [Node] = [
...
-        name: "eachKeyword",
-        deprecatedName: "each",
-        kind: .token(choices: [.keyword(.each)]),
-        nameForDiagnostics: "parameter pack specifier",
+        name: "specifier",
+        kind: .token(choices: [
+          .keyword(.each),
+          .keyword(.let)
+        ]),
+        nameForDiagnostics: "specifier",
```
Comment (ahoppen) [inline CodeGeneration/Sources/SyntaxSupport/GenericNodes.swift:101]: Could you add `deprecatedName: "eachKeyword"` so we get compatibility accessors for `eachKeyword`?

File: Sources/SwiftParser/Declarations.swift [hunk 2]
```diff
@@ -431,11 +431,18 @@ extension Parser {
...
-        var each = self.consume(if: .keyword(.each))
+        var specifier: Token? = nil
+
+        // Parse the 'each' keyword for a type parameter pack 'each T' or a
+        // 'let' keyword for a value parameter 'let N: Int'.
+        if let each = self.consume(if: .keyword(.each)) {
+          specifier = each
+        } else if let `let` = self.consume(if: .keyword(.let)) {
+          specifier = `let`
+        }
```
Comment (ahoppen) [inline Sources/SwiftParser/Declarations.swift:442]: You can just do 

```suggestion
        let specifier = self.consume(if: .keyword(.each), .keyword(.let))
```

**Review by ahoppen — COMMENTED** [review]
Thanks for the test cases as well.

**Comment by ahoppen** [comment]
Sorry, forgot in the last review: Could you also add a release note entry to `Release Notes/601.md` for the rename `eachKeyword` -> `specifier`?

**Review by ahoppen — APPROVED** [review]

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please test Windows

**Pull request merged**